spec/services/flags/create_flag_spec.rb

```diff
@@ -1,105 +1,78 @@
 # frozen_string_literal: true
 
 RSpec.describe(Flags::CreateFlag) do
-  subject(:result) do
-    described_class.call(
-      guardian: current_user.guardian,
-      name: name,
-      description: description,
-      applies_to: applies_to,
-      require_message: require_message,
-      enabled: enabled,
-    )
+  describe described_class::Contract, type: :model do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:description) }
+    it { is_expected.to validate_length_of(:name).is_at_most(Flag::MAX_NAME_LENGTH) }
+    it { is_expected.to validate_length_of(:description).is_at_most(Flag::MAX_DESCRIPTION_LENGTH) }
+    it { is_expected.to validate_inclusion_of(:applies_to).in_array(Flag.valid_applies_to_types) }
   end
 
-  let(:name) { "custom flag name" }
-  let(:description) { "custom flag description" }
-  let(:applies_to) { ["Topic"] }
-  let(:enabled) { true }
-  let(:require_message) { true }
+  describe ".call" do
+    subject(:result) { described_class.call(**params, **dependencies) }
 
-  context "when user is not allowed to perform the action" do
-    fab!(:current_user) { Fabricate(:user) }
-
-    it { is_expected.to fail_a_policy(:invalid_access) }
-  end
-
-  context "when title is not unique" do
-    fab!(:current_user) { Fabricate(:admin) }
-    fab!(:flag) { Fabricate(:flag, name: "custom flag name") }
-
-    it { is_expected.to fail_a_policy(:unique_name) }
-
-    after { Flag.destroy_by(name: "custom flag name") }
-  end
-
-  context "when applies to is invalid" do
```

Comment:
To test a contract, it’s easier to do it like on lines 4 to 9. (we can use shoulda matchers making its specs very easy to write)